Starwalker-Class!

spockye opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
![image](https://user-images.githubusercontent.com/79304582/178705941-a959f05a-4e85-47f1-a08c-129bac920304.png)


![2022 07 13-11 18 11](https://user-images.githubusercontent.com/79304582/178698522-0f74694d-fd3c-4c8d-ab01-f0a047fac660.png)



![image](https://user-images.githubusercontent.com/79304582/178163242-7c034bf2-1112-46e6-9f83-e6124036e915.png)


this pr adds a new ship, called the Starwalker class! this ship's purpose is as a small carrier used for piracy.  it comes equipped with a small hangar, two emitter bays, a well-stocked medbay, and a decently filled armoury.  
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
we currently only have ONE pirate ship, and i want to change that! 
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Starwalker class pirate carrier
add: a new pirate subtype of the meat hook
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

spockye committed:
adds a new pirate ship

Apostolis! (it sounds greek ig

PiperDoots commented:
Can you add a picture of the ship please

spockye commented:
> Can you add a picture of the ship please

was waiting for mapdiff to finish rendering!

Apogee-dev commented:
The bridge still needs a chair or two and I might suggest mixing up the leading edge of the ship a little since it’s totally flat right now. Maybe just extend the middle section of the front one tile forward from the side hangar or something. Other than that… I can’t say I’m a fan of all the tinyfan airlocks here. I dislike tinyfans in general but at the very least they should be restricted to cargo bays, not regular airlocks. If at all possible, those should be AAC airlocks instead.

spockye commented:
> can’t say I’m a fan of all the tinyfan airlocks here. I dislike tinyfans in general but at the very least they should be restricted to cargo bays, not regular airlocks. If at all possible, those should be AAC airlocks instead.

theres only one tinyfan airlock, and its the one at cargo. the ones at the engineering bays dont have ones, and are boarded up.

spockye committed:
tries to fix sec officers, and misc changes

adds space areas
moves bay one down
adds chair

Fikou commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -419,6 +419,15 @@
 	heat_protection = CHEST|GROIN|ARMS|LEGS
 	cold_protection = CHEST|GROIN|ARMS|LEGS
 
+/obj/item/clothing/suit/jacket/leather/duster/armoured
```
this needs a different name or icon so you can tell whether its armored or not

spockye replied:
changed it to "Armoured leather duster"

Fikou commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -281,6 +281,14 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
+	name = "First mate(pilot)"
```
make this normal

spockye replied:
what do you mean with normal?

Fikou replied:
it does not follow the other outfits in "grammar"

Fikou commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -281,6 +281,14 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
+	shoes = /obj/item/clothing/shoes/jackboots
+	glasses = /obj/item/clothing/glasses/sunglasses/big
+	suit = /obj/item/clothing/suit/jacket/leather/duster/armoured
+	backpack_contents = list(/obj/item/storage/box/survival/security=1)
```
survival security box is already included by the box variable, this will make them spawn with 2

spockye replied:
ooh, alright. i'll get that fixed

spockye committed:
fixes!

nerds' all of ya!

spockye committed:
more misc changes

adds some plasma, moves some thingies

spockye committed:
connects air tank

oopsies!

spockye committed:
stop turning

i know where you live, IN THE CODE

spockye committed:
adds prefix and namelist

spockye commented:
if anyone could show me how to force the ship to always land in the same direction, it would be greatly appreciated. the current pod doors I'm using don't have automated directional sprites, and break upon landing. 
(it also generally just looks worse when it lands in the unintended direction)

Latentish reviewed (approved):
TMable? No not yet there are requests ahead of this. But it looks good so far.

The original _**Starwalker...**_
![image](https://user-images.githubusercontent.com/55075747/178361836-10ef268a-0e8b-4d36-9097-d25eb5658233.png)

Apogee-dev commented on code/modules/clothing/suits/miscellaneous.dm:
```diff
@@ -419,6 +419,15 @@
 	heat_protection = CHEST|GROIN|ARMS|LEGS
 	cold_protection = CHEST|GROIN|ARMS|LEGS
 
+/obj/item/clothing/suit/jacket/leather/duster/armoured
```
Is this strictly necessary? We have a black armored duster already, which is a subtype of the armor vest (and therefore also benefits from the suit storage permissions that come with it). Having a brown version (admittedly with weaker armor) seems redundant. Additionally, if you do keep it, change all new instances of "armoured" to "armored" for consistency. We may as well standardize at some point and most other clothing items use the american spelling.

Apogee-dev commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 
+/datum/outfit/job/security/pirate/pilot
+	name = "First Mate (pilot)"
+	uniform = /obj/item/clothing/under/costume/singer/yellow
```
This seems like an odd choice in uniform for a normal job, considering it's a Vocaloid costume. It also has frankly ugly sprites and isn't likely to get resprited anytime soon.

Apogee-dev commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	id = /obj/item/card/id/syndicate_command/crew_id
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
```
Why is this a subtype of security officer? "first mate" positions are typically a subtype of HoP and treated as officer ranks. You might want to use the existing pirate HoP outfit for this slot.

Apogee-dev commented on _maps/configs/pirate_apostolis:
```diff
@@ -0,0 +1,37 @@
+{
+	"map_name": "apostolis-class carrier",
```
Minor gripe, but the name doesn't seem very fitting for a pirate ship. Greek tends to come off as having a bit more pretense than usual and pirates are anything but. I'd suggest something explicitly pirate-related like nautical terminology or the name of a historical port (Tortuga, Port Royal, and so on).

spockye replied:
i chose the name after searching up libertia, and choosing one of its "sister names" wanting to keep the pirate ships in the same naming category. though I'll try to find a new one

Apogee-dev commented:
Other than the issues raised in the review, layout looks good with one exception. The boarded up rear airlocks should either have tinyfans (not ideal but preferable to nothing) or be turned into AAC airlocks. In their current state they're impossible to use without leaking atmos into the engine bays (and ultimately the rest of the ship), and they're the only way out of the back of the ship so they _will_ be used.

spockye commented:
for the first mate role, he doesn't act like a "hop", but more as the pod pilot. I chose his set of clothing after messing around in-game, found something that fit well together, and gave a look that I felt fitting. alone Vocaloid costume looks pretty bad, but it also looks decent as a "tattered pirate shirt" when it's under another set of clothing.  Same with the leather duster. i was thinking about using the Hos version, but it just didnt feel very piratey, unlike the leather duster. (I'll get working on Americanizing it ;()

spockye committed:
turns the back engi rooms into airlocks

also changes the parrots name to "Pepper"!

Apogee-dev commented:
> for the first mate role, he doesn't act like a "hop", but more as the pod pilot. I chose his set of clothing after messing around in-game, found something that fit well together, and gave a look that I felt fitting. alone Vocaloid costume looks pretty bad, but it also looks decent as a "tattered pirate shirt" when it's under another set of clothing. Same with the leather duster. i was thinking about using the Hos version, but it just didnt feel very piratey, unlike the leather duster. (I'll get working on Americanizing it ;()

If he's acting as the pilot I'd choose a different name and remove the officer flag, then. You could call the job "Flyboy" or "Pod Jockey" or something similarly informal. I'd also recommend adding a proper first-mate-as-officer role just to ensure the ship actually has two officers (including the captain). Honestly I also think you should go with the black duster or an unarmored brown duster. Snowflaking in an armored brown variant just for this one job just doesn't seem like the greatest idea to me.

spockye committed:
adds some random dingies

changes the pod to basic
adds some hull platings
some more medical stuff: herb healing/1xadvanced stuture
skub/tape/sand
changes the bed type to the old version

spockye committed:
Misc changes

Removes the leather jacket :(

adds a HOP role

changes pilot name to "Pod Jockey"

changes to leather jacket from armoured

spockye committed:
misc balance

adds fire alarms

adds a winchester

adds a 38. revolver

removes 1 medkit

adds more ammo

adds lightswitches

spockye committed:
name change + balance

Removes the meat hook
removes junk weapons
two riot shutguns
two winchesters
1 38. revolver

Sylphily committed:
Makes a pirate hool

for a friend

spockye committed:
Merge pull request #1 from someguyjust/test

Makes a pirate hook

spockye committed:
adds a missing catwalk

boop di boop di boop di boop di-

spockye committed:
changes the outside zone to be cargo

Apogee-dev commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -539,6 +539,16 @@
+//pirate version
+/obj/item/gun/magic/hook/pirate
+	name = "boarding hook"
+	desc = "A long metal chain, with a large hook connected to the end. Once used to board ancient Merchant vessels, now used to drag un-suspecting Planet-crabs closer to you."
```
i suspect "planet-crabs" might be a direct translation or a modified idiom that didn't quite make it to english very well.  You might want to make the description a little more clearly descriptive, particularly when it comes to how this differs from a normal meat hook. Suggested desc below.

desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."

Apogee-dev commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -539,6 +539,16 @@
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
+	user.changeNext_move(CLICK_CD_MELEE * 1)//quick to swing. 15 force can be quite something with this attack frequency.
```
the *1 is unnecessary here, multiplying by one doesn't do anything. just make it (CLICK_CD_MELEE).

Sylphily replied:
whoops my bad lol

Apogee-dev commented on _maps/shuttles/shiptest/pirate_starwalker.dmm:
```diff
@@ -0,0 +1,4077 @@
+	pixel_y = 6
+	},
+/obj/item/gun/ballistic/automatic/zip_pistol,
+/obj/item/gun/ballistic/automatic/zip_pistol,
```
suggested armory loadout: 2x pump shotgun, 2x .38 revolver, 2x winchester. Nix the zipguns and improv shotguns and replace the shivs with hunting knives. They're basically just kitchen knives that butcher faster. The crew can print kitchen knives anyway so they'll just replace the shivs immediately.

Apogee-dev commented on _maps/configs/pirate_apostolis:
```diff
@@ -0,0 +1,47 @@
+{
+	"map_name": "Starwalker-class carrier",
+	"prefix": "ISV",
+	"namelists": ["BEASTS", "WEAPONS"],
```
Add the pirate namelist. What's the point of a pirate ship that doesn't use the funny pirate names?

spockye replied:
i couldnt find a pirate prefix... is it just called "pirate" ?

Apogee-dev replied:
Namelist, not prefix. It’s the same one the Libertatia uses, “PIRATES”.

Apogee-dev commented on _maps/configs/pirate_apostolis:
```diff
@@ -0,0 +1,47 @@
+{
+	"map_name": "Starwalker-class carrier",
```
"Starwalker" is actually already a standard ship name in the general namelist. This may cause some confusion.

Latentish replied:
I dont think this is a huge issue tbh

Apogee-dev replied:
It's also a bit jokey and doesn't really sound appropriate/piratey, tbh, but those are personal gripes.

spockye committed:
oopsies

added the armourt change to the wrong version

spockye committed:
Merge branch 'Pirate-carrier' of https://github.com/spockye/Shiptest into Pirate-carrier

Sylphily commented:
that outfit for that one role is a bit funky

Sylphily committed:
Update necropolis_chests.dm

spockye committed:
Merge pull request #2 from someguyjust/test

Changes hook wise

spockye committed:
silly string wiring!

im not insane, im not insane, im not insane, im not insane, im not insane, im not insane, im not insane, im not insane, im not insane.

also increases volume in the oxy tank, to make up for the huge airlocks

spockye committed:
Merge branch 'Pirate-carrier' of https://github.com/spockye/Shiptest into Pirate-carrier

spockye committed:
updates the json file

some leftover "apostolosis"

Apogee-dev commented on _maps/shuttles/shiptest/pirate_starwalker.dmm:
```diff
@@ -0,0 +1,4047 @@
+	},
+/obj/machinery/atmospherics/components/unary/tank/air{
+	piping_layer = 2;
+	volume = 25000
```
Please don't varedit statistics like volume. If it's really necessary, you can add another air tank.

Apogee-dev commented on _maps/shuttles/shiptest/pirate_starwalker.dmm:
```diff
@@ -0,0 +1,4047 @@
+/obj/item/gun/ballistic/shotgun/winchester,
+/obj/item/gun/ballistic/revolver/detective,
+/obj/item/gun/ballistic/shotgun/riot,
+/obj/item/gun/ballistic/shotgun/riot,
```
Riot shotguns might be overkill- while balance was different at the time, riot shotties were a major reason for the Libertatia's early nerfs. Also keep in mind they come pre-loaded with beanbags. Use the normal, 4-shot variety. Also, don't forget to re-add the boarding hook. You have it in the code but not placed on the ship.

Apogee-dev commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -539,6 +539,16 @@
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
+	user.changeNext_move(CLICK_CD_MELEE)//quick to swing. 15 force can be quite something with this attack frequency.
```
Don't forget to remove the comment here, it's copied from the base meat hook and doesn't apply to the boarding hook.

spockye committed:
misc changes

removes riot shotguns
adds db shotguns
adds the pirate hook

spockye committed:
removes waste text!

Apogee-dev commented on _maps/shuttles/shiptest/pirate_starwalker.dmm:
```diff
@@ -0,0 +1,4047 @@
+	},
+/obj/item/gun/magic/hook/pirate,
+/obj/item/gun/ballistic/shotgun/doublebarrel,
+/obj/item/gun/ballistic/shotgun/doublebarrel,
```
db shotguns swing back into not having quite enough firepower. just replace with /obj/item/gun/ballistic/shotgun/lethal. everything else is good

spockye committed:
changes shottie!! type

also some more ingredients

Fikou commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -539,6 +539,16 @@
+	ammo_type = /obj/item/ammo_casing/magic/hook/bounty
+
+/obj/item/gun/magic/hook/melee_attack_chain(mob/user, atom/target, params)
+	..()
```
You need to set return value here.

Fikou commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -281,6 +281,13 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	backpack_contents = list(/obj/item/storage/box/survival/syndie=1)
 
+/datum/outfit/job/security/pirate/pilot
+	name = "Pod Jockey (pilot)"
```
```suggestion
	name = "Pod Jockey (Pilot)"
```

spockye committed:
Update code/modules/jobs/job_types/security_officer.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

Latentish reviewed (changes requested):
![image](https://user-images.githubusercontent.com/55075747/179388444-b0927d49-925f-4483-bcb1-ea42317c2ba1.png)
You have a redundant (assumedly) copy of the ship from before you renamed it. Please remove it if it is not needed.

After that I'd like to see this get TM'd to see how it plays in game and to find more issues.

spockye committed:
Removes the apostilis! 

redundant file

Sylphily committed:
return

Sylphily committed:
Merge branch 'Pirate-carrier' into tester

spockye committed:
Merge pull request #3 from Sylphily/tester

adds the return thingy

spockye commented:
😔

Latentish commented:
L😔

Sylphily commented:
could some one explain what the fuck is failing?

ZephyrTFA commented on code/modules/mining/lavaland/necropolis_chests.dm:
```diff
@@ -539,6 +539,17 @@
+	. = ..()
+	user.changeNext_move(CLICK_CD_MELEE)//makes the hook unquick to swing
+	return .
+  
```
```suggestion
/obj/item/gun/magic/hook/pirate
	name = "boarding hook"
	desc = "A long metal chain, with a heavy hook connected to the end that can make deadly blows in close quarters. Once used to board ancient merchant vessels, now used to drag unsuspecting victims into close range."
	ammo_type = /obj/item/ammo_casing/magic/hook/bounty

/obj/item/gun/magic/hook/pirate/melee_attack_chain(mob/user, atom/target, params)
	. = ..()
	user.changeNext_move(CLICK_CD_MELEE)//makes the hook unquick to swing
	return .
 
```

spockye replied:
Something still seems to be failing...

🥲

spockye committed:
Update code/modules/mining/lavaland/necropolis_chests.dm

Co-authored-by: Zephyr <12817816+ZephyrTFA@users.noreply.github.com>

ZephyrTFA commented:
Actually reading the Linters might help you

ZephyrTFA commented:
```
ERROR: found door names without proper upper-casing. 
```

Sylphily committed:
weird standards

spockye committed:
Merge pull request #5 from Sylphily/Starwalker-door

dummy V2 (linter massacare)

spockye commented:
Oh dear

Sylphily commented:
lets just not do the decals for now

spockye commented:
Im furious, and sad. 

I'll do this another time
